internal/decomposer/decomposer.go

```diff
@@ -34,7 +34,7 @@ type ExecutionDetails struct {
 }
 
 type IDecomposer interface {
-	Execute(playbook cacao.Playbook) (*ExecutionDetails, error)
+	Execute(playbook cacao.Playbook, details chan ExecutionDetails) (*ExecutionDetails, error)
 }
 
```

Comment:
I think we need a new interface to leave this one as is ExecuteAsync and Execute so the Playbook action step can be untouched.